### ISSUE TITLE
PDF Export on Windows not supporting custom size (2020 regression)

### ISSUE THREAD

**Issue by Eneroth3** [SketchUp/api-issue-tracker#519]
In both SketchUp 2018 and 2019 this code snippet generates PDFs at various sizes. In SU 2020 they are all identical.

```ruby
dir = UI.select_directory

Sketchup.active_model.export(
  "#{dir}/100.0.pdf",
  {
    height_units: Length::Inches,
    window_height: 100.0
  }
)
Sketchup.active_model.export(
  "#{dir}/10.0.pdf",
  {
    height_units: Length::Inches,
    window_height: 10.0
  }
)
Sketchup.active_model.export(
  "#{dir}/1.0.pdf",
  {
    height_units: Length::Inches,
    window_height: 1.0
  }
)
Sketchup.active_model.export(
  "#{dir}/0.1.pdf",
  {
    height_units: Length::Inches,
    window_height: 0.1
  }
)
```

Have the keywords changed recently? I've also tried image_height, imageHeight and any other keyword I can think of but to no avail.

**Comment by sketchupbot** [comment]
Logged as: SU-46985

**Comment by Eneroth3** [comment]
I am extremely confused. It works today. Not sure why it didn't the other day. Must have had the files locked or some other error. Closing this issue.

**Comment by Eneroth3** [comment]
I'm not mad! I've been able to successfully reproduce this with the exact code above, **but only in parallel projection**.

It appears to also be an issue in SketchUp 2019, but with a slightly different behavior.

When running the script in 2019 I get this result when opening one of the resulting PDFs in Adobe Reader. Weird warning and icon not centered in the image.

![2020-08-24_13h35_33](https://user-images.githubusercontent.com/5631725/91040726-fd310400-e60e-11ea-8613-51450c0e9494.png)
![2020-08-24_13h35_39](https://user-images.githubusercontent.com/5631725/91040733-01f5b800-e60f-11ea-8d64-a8784d1854a6.png)
![2020-08-24_13h42_47](https://user-images.githubusercontent.com/5631725/91041088-b4c61600-e60f-11ea-9a90-d81f5511c04b.png)

When running the same code in 2020, I get this result. The icon is now centered as it should be.

![2020-08-24_13h42_07](https://user-images.githubusercontent.com/5631725/91041032-9fe98280-e60f-11ea-99e6-36f8d0fd7d82.png)
![2020-08-24_13h42_02](https://user-images.githubusercontent.com/5631725/91041038-a1b34600-e60f-11ea-83a4-564e3b42732b.png)

In both 2020 and 2019, the image height value is completely ignored and all 4 PDFs get the same height (although this value differs between the two versions).

---

For now I can probably use a perspective with a tiny fov and get something that mimics a parallel projection well enough not to be noticed, but in the long run it'd be nice to have this fixed. 

I'd expect the image height parameter to set the scale for the generated 2d view, regardless of it being a perspective or parallel projection. In addition to my little icon pipeline, this would be highly useful to output scale drawings.

**Comment by Eneroth3** [comment]
It appears to be a separate set of parameters used in parallel projection. With no proper error messages it is hard to figure this out, especially with no access to the code and no ability to read when what parameters are actually referenced.

More code examples would be very useful, but ideally there should also be a Ruby warning (or possibly an exception but it could break old extensions) when passing image_height for a parallel projection view, length_in_model and length_in_drawing when in perspective etc.

**Comment by Matherone** [comment]
Seconding Eneroth3's request. Fixing this would be useful. 

It would enable e.g. a Sketchup extension for dependably turning Sketchup models into vector icons for use in Sketchup extensions - doogfooding at its finest.
